src/firefly/java/edu/caltech/ipac/firefly/server/servlets/AnyFileDownload.java

```diff
@@ -101,4 +112,16 @@ private static void trackProgress(HttpServletRequest req, SearchServices.Downloa
     }
 
     public static Cache getCache() { return UserCache.getInstance(); }
+
+    private static File retrieveHiPSData(String urlStr) throws Exception {
+
+        URL url= new URL(urlStr);
+
+        File dir= new File(ServerContext.getHiPSDir(),new File(url.getHost()+"/"+url.getPath()).getParent());
+        if (!dir.exists()) dir.mkdirs();
+        File targetFile= new File(dir, new File(url.getFile()).getName());
+//        if (targetFile.canRead()) return targetFile;
```

Comment:
commented because URLDownload takes care of the caching?

Reply:
`URLDownload` uses `HTTP_NOT_MODIFIED`. That means it checks it there is a more recent  file and downloads it it there is.